addons/website_event/models/event_event.py

```diff
@@ -267,11 +309,14 @@ def _update_website_menus(self, menus_update_by_field=None):
             elif event.website_menu and not event.menu_id:
                 root_menu = self.env['website.menu'].sudo().create({'name': event.name, 'website_id': event.website_id.id})
                 event.menu_id = root_menu
-            if event.website_menu and (not menus_update_by_field or event in menus_update_by_field.get('website_menu')):
-                for name, url, xml_id, menu_sequence, menu_type in event._get_website_menu_entries():
-                    event._create_menu(menu_sequence, name, url, xml_id, menu_type=menu_type)
             if event.menu_id and (not menus_update_by_field or event in menus_update_by_field.get('community_menu')):
                 event._update_website_menu_entry('community_menu', 'community_menu_ids', '_get_community_menu_entries')
+            if event.menu_id and (not menus_update_by_field or event in menus_update_by_field.get('introduction_menu')):
+                event._update_website_menu_entry('introduction_menu', 'introduction_menu_ids', '_get_introduction_menu_entries')
+            if event.menu_id and (not menus_update_by_field or event in menus_update_by_field.get('location_menu')):
+                event._update_website_menu_entry('location_menu', 'location_menu_ids', '_get_location_menu_entries')
+            if event.menu_id and (not menus_update_by_field or event in menus_update_by_field.get('register_menu')):
+                event._update_website_menu_entry('register_menu', 'register_menu_ids', '_get_register_menu_entries')
 
     def _update_website_menu_entry(self, fname_bool, fname_o2m, method_name):
```

Comment:
This is a bit strange.

Now that every entry of "_get_website_menu_entries" has its own "menu_type", maybe we could give "menu_type" to this method instead of "method_name" and it will retrieve the corresponding entry using "_get_website_menu_entries" and filtering on the matching menu_type?

It feels like unnecessary overhead to have all those "_get_XXX_menu_entries".

Reply:
Indeed. Note that it is still not a one to one relationship (track has 2 menus, tracks and agenda), but we can already merge all social menus definition into a single one and find back the right one based on type.